File: engine/model/src/main/scala/tools/packet/guild/GuildMemberLevelJobUpdate.scala
package tools.packet.guild

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GuildMemberLevelJobUpdate(private var _guildId: Int, private var _characterId: Int, private var _level: Int,
                                private var _jobId: Int) extends PacketInput {
  def guildId: Int = _guildId

  def characterId: Int = _characterId

  def level: Int = _level

  def jobId: Int = _jobId

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/GuildMemberLevelJobUpdate.java
package tools.packet.guild;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GuildMemberLevelJobUpdate(Integer guildId, Integer characterId, Integer level,
                                        Integer jobId) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}